src/frontend/src/binder/relation/join.rs

```diff
@@ -93,9 +103,58 @@ impl Binder {
                 }
                 bound_expr
             }
-            JoinConstraint::Using(_columns) => {
-                return Err(ErrorCode::NotImplemented("USING".into(), 1636.into()).into())
+            JoinConstraint::Using(columns) => {
+                let mut columns_iter = columns.into_iter();
+                let first_column = columns_iter.next().unwrap();
+                let mut binary_expr = Expr::BinaryOp {
+                    left: Box::new(Expr::CompoundIdentifier(vec![
+                        left_table.clone().unwrap(),
+                        first_column.clone(),
+                    ])),
+                    op: BinaryOperator::Eq,
+                    right: Box::new(Expr::CompoundIdentifier(vec![
+                        right_table.clone().unwrap(),
+                        first_column,
+                    ])),
+                };
+                for column in columns_iter {
+                    binary_expr = Expr::BinaryOp {
+                        left: Box::new(binary_expr),
+                        op: BinaryOperator::Eq,
+                        right: Box::new(Expr::BinaryOp {
+                            left: Box::new(Expr::CompoundIdentifier(vec![
+                                left_table.clone().unwrap(),
+                                column.clone(),
```

Comment:
Unfortunately this does not work in all cases:
```
create table ab(a int, b int);
create table bc(b int, c int);
create table ca(c int, a int);
create table abc (a int, b int, c int);
```
The following works in PostgreSQL (`test=#`) but not by this PR (`dev=>`):
```
test=# select * from ab join bc using(b) join ca using(c);
 c | b | a | a 
---+---+---+---
(0 rows)

dev=> explain select * from ab join bc using(b) join ca using(c);
ERROR:  Item not found: Invalid column: c
```

How does PG know to find column `c` in table `bc` rather than in table `ab`? It does not know:
```
test=# select * from abc join bc using(b) join ca using(c);
ERROR:  common column name "c" appears more than once in left table
```

When binding `join-on`, we populate the same context with columns from both left side and right side, and then resolve names in this context.

But for `join-using`, it seems we have to separate left side and right side, and bind it in each side without qualifying it with table names.

I am not sure how much this is related to the refactor I mentioned earlier. But both of us has underestimated the edge cases that can happen during join. Let's make sure we understand the expected behavior first.